NTupleAnalyzerHH_nanoAOD.py

```diff
@@ -1290,24 +1503,30 @@ def LooseIDMuons(T,_met,variation,isdata):
                 # Require Loose muon flag
 	        Pass *= T.Muon_looseId[n] > 0
 
+	        #fixme these two are equivalent to IsLooseMuon #?
+		#Pass *= T.MuonIsPF[n]
+	        #Pass *= (T.MuonIsGlobal[n] or T.MuonIsTracker[n])
+
 		# A preliminary pT cut.
 		Pass *= (_MuonPt[n] > 8)#fixme offline cuts are 20(10)
 
 		# Eta requirement
 	        Pass *= abs(T.Muon_eta[n])<2.4
 
 		# Isolation condition using combined PF relative isolation - 0.25 for loose (98% efficiency), 0.15 for tight (95% efficiency)
-	        correctedIso = T.Muon_pfRelIso04_all[n] 
+	        correctedIso = T.Muon_pfRelIso04_all[n]
 		# Don't apply isolation for QCD studies
 		if nonisoswitch != True:
-			Pass *= correctedIso < 0.25
+			Pass *= correctedIso/_MuonPt[n] < 0.25
 
 		# Propagate MET changes if undergoing systematic variation
 		if (Pass):
 			NewMu = TLorentzVector()
 			OldMu = TLorentzVector()
-			NewMu.SetPtEtaPhiM(_MuonPt[n],T.Muon_eta[n],T.Muon_phi[n],0)
-			OldMu.SetPtEtaPhiM(T.Muon_pt[n],T.Muon_eta[n],T.Muon_phi[n],0)
+			NewMu.SetPtEtaPhiM(_MuonPt[n],T.Muon_eta[n],T.Muon_phi[n],T.Muon_mass[n])
```

Comment:
This looks like a problem in the indentation compared to the old version, please check.